pkg/apis/core/v1beta1/types_shoot.go

```diff
@@ -694,6 +694,12 @@ type KubeProxyConfig struct {
 	// defaults to IPTables.
 	// +optional
 	Mode *ProxyMode `json:"mode,omitempty" protobuf:"bytes,2,opt,name=mode,casttype=ProxyMode"`
+	// Enabled indicates whether kube-proxy should be deployed or not.
+	// Depending on your networking provider extensions switching kube-proxy off might be rejected.
```

Comment:
Does it make sense to add a note for networking extensions to https://github.com/gardener/gardener/blob/master/docs/extensions/network.md, that they should
- check this field and act accordingly upon it during reconciliation
- fail reconciliation if proxy-less is not supported
- optionally add a validation component for the garden cluster to prevent misconfiguration by users?

Reply:
Added two sentences to `network.md` as also requested by @rfranzke. Please take a look and provide feedback in case this is not sufficient.

Reply:
For posterity:
@ScheererJ and I agreed offline, that the contract description should be enhanced, that networking extensions have to look at the `kubeproxy.enabled` field (by checking the `Cluster` resource) in order to configure the data plane accordingly.
We also agreed, that the reconciliation of the `Network` extension resource shall fail, in case an invalid/unsupported configuration is given, so that we surface configuration errors leading to broken service routing before applying the configuration.
An validation component in the garden cluster prevent such configuration errors in the first place is optional, but makes sense. Still, the reconciliation should fail, to surface errors also in the case, that such a validation component is not deployed by operators.

Reply:
Adapted documentation in `network.md` to clarify the contract around `kube-proxy`.